src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -32,6 +33,12 @@ internal class ObjectWriter : IDisposable
 
         private const string NativeObjectWriterFileName = "objwriter";
 
+        // Target platform ObjectWriter is instantiated for.
+        private TargetDetails _targetPlatform;
```

Comment:
The EmitSymbolDefinition statement was unrequired - the idea is to not pass the details as arguments.